frontend/src/components/Map.tsx

```diff
@@ -37,20 +54,39 @@ export function Map(props: ViewProps): React.ReactElement<ViewProps> {
       initialRegion={GOLDEN}
       showsUserLocation={true}
       showsMyLocationButton={false}
+      mapPadding={padding}
       // followsUserLocation is only available on iOS maps, and isn't very cooperative anyway.
       // Reimplement it ourselves.
       onUserLocationChange={ event => { panToLocation(event.nativeEvent.coordinate) }}      
-      onRegionChange={(_region, details) => {
-        // If the user is panning around, we don't want to snap back to their location. Note that we
-        // make sure to exclude camera pans from this to avoid disabling location following at soon
-        // as it changes.
-        if (details.isGesture ?? true) {
-          setUserRegionChanged(true) 
-        }
-      }} />
+      onRegionChange={(_region, details) => { handleRegionChange(details) }} />
   )
 }
 
+/**
+ * The props for the {@interface Map} component.
+ */
+export interface MapProps {
+  /** The {@interface Insets} to apply to the map. */
+  insets?: Insets
+}
+
+/**
+ * The insets to apply to the {@interface Map} component when it will be obscured by
+ * other components. This will shift map components like the google logo, and the way
+ * that the map camera will pan around. Note that the component will already be insetting
+ * by the status bar, so you don't need to include that in your insets.
+ */
+export interface Insets {
+  /** The amount of space to inset from the top of the map. */
+  top: number,
+  /** The amount of space to inset from the left of the map. */
+  left: number,
+  /** The amount of space to inset from the bottom of the map. */
+  bottom: number,
+  /** The amount of space to inset from the right of the map. */
+  right: number
+}
+
 const styles = StyleSheet.create({
```

Comment:
should this be in a seperate file? ik its not super large right now so fine to keep but we should probably move this out if it gets to be any larger